Gems/Atom/Asset/Shader/Code/Source/Editor/ShaderVariantAssetBuilder.cpp

```diff
@@ -1140,10 +1141,137 @@ namespace AZ
                 }
             }
 
+            if (shaderVariantInfo.m_enableRegisterAnalysis && strcmp(creationContext.m_shaderPlatformInterface.GetAPIName().GetCStr(), "vulkan") == 0)
+            {
+                AZ::IO::FixedMaxPath projectBuildPath = AZ::Utils::GetExecutableDirectory();
+                projectBuildPath = projectBuildPath.RemoveFilename(); // profile
+                projectBuildPath = projectBuildPath.RemoveFilename(); // bin
+
+                AZStd::string spirvPath;
+                AZStd::string spirvFile = AZStd::string::format(
+                    "%s_vulkan_%u.spirv.bin", creationContext.m_shaderStemNamePrefix.c_str(), shaderVariantInfo.m_stableId);
+                AZ::StringFunc::Path::Join(creationContext.m_tempDirPath.c_str(), spirvFile.c_str(), spirvPath, true, true);
+
+                AZStd::string rgaCommand = AZStd::string::format(
+                    "-s vk-spv-offline --isa ./disassem_%u.txt --livereg ./livereg_%u.txt --asic %s",
+                    shaderVariantInfo.m_stableId,
+                    shaderVariantInfo.m_stableId,
+                    shaderVariantInfo.m_asic.c_str());
+
+                AZStd::string command = AZStd::string::format("%s%s %s %s", projectBuildPath.c_str(), RgaExePath, rgaCommand.c_str(), spirvPath.c_str());
+                AZ_TracePrintf(ShaderVariantAssetBuilderName, "Rga command %s", command.c_str());
+
+                AZStd::string failMessage;
+                if (!LaunchRadeonGPUAnalyzer(command, creationContext.m_tempDirPath, failMessage))
+                {
+                    return AZ::Failure(failMessage);
+                }
+
+                // add rga output to the by product list
+                outputByproducts->m_intermediatePaths.insert(AZStd::string::format("./%s_disassem_%u_frag.txt",shaderVariantInfo.m_asic.c_str(), shaderVariantInfo.m_stableId));
+                outputByproducts->m_intermediatePaths.insert(AZStd::string::format("./%s_livereg_%u_frag.txt", shaderVariantInfo.m_asic.c_str(),shaderVariantInfo.m_stableId));
+            }
+            
+
             Data::Asset<RPI::ShaderVariantAsset> shaderVariantAsset;
             variantCreator.End(shaderVariantAsset);
             return AZ::Success(AZStd::move(shaderVariantAsset));
         }
 
+        bool ShaderVariantAssetBuilder::LaunchRadeonGPUAnalyzer(AZStd::string& command, const AZStd::string& workingDirectory, AZStd::string& failMessage)
+        {
+            AzFramework::ProcessLauncher::ProcessLaunchInfo processLaunchInfo;
+            processLaunchInfo.m_commandlineParameters.emplace<AZStd::string>(command);
```

Comment:
I suggest using the AZStd::vector<AZStd::string> version for the command line parameters as that can properly deal with paths that have spaces in them.
For example if the user had a project at `C:\Users\My User Name\My Documents\TestProject`, then this command will have issues.

So to avoid that, this function should accept an `AZStd::vector<AZStd::string> command` and then move it into the processLaunchInfo.

```suggestion
        bool ShaderVariantAssetBuilder::LaunchRadeonGPUAnalyzer(AZStd::vector<AZStd::string> command, const AZStd::string& workingDirectory, AZStd::string& failMessage)
        {
            AzFramework::ProcessLauncher::ProcessLaunchInfo processLaunchInfo;
            processLaunchInfo.m_commandlineParameters.emplace<AZStd::vector<AZStd::string>>(AZStd::move(command));
```